### PR TITLE
dtest for CASSANDRA-10559

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

**Comment by snazy** [comment]
The dtest should be safe to merge even before 10559

File: native_transport_ssl_test.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 from dtest import Tester
 from cassandra import ConsistencyLevel
 from cassandra.cluster import NoHostAvailable
-from tools import generate_ssl_stores, putget, since
+from tools import generate_ssl_stores, putget, since, require
```
Comment (ptnapoleon) [inline native_transport_ssl_test.py]: No need to import `require` here

File: native_transport_ssl_test.py [hunk 2]
```diff
@@ -35,6 +35,23 @@ def connect_to_ssl_test(self):
         session = self.patient_cql_connection(node1, ssl_opts={'ca_certs': os.path.join(self.test_path, 'ccm_node.cer')})
         self._putget(cluster, session)
 
+    @since('3.0')
+    def connect_to_ssl_optional_test(self):
+        """
+        Connecting to SSL optional native transport port must be possible with SSL and non-SSL native clients
```
Comment (ptnapoleon) [inline native_transport_ssl_test.py]: If this test is primarily for 10559, can you add this line in the comment block:
`@jira_ticket CASSANDRA-10559`

**Comment by ptnapoleon** [comment]
@snazy can you rebase this, please? Thanks

**Comment by snazy** [comment]
Sure, rebased it

**Comment by ptnapoleon** [comment]
Thanks

**Comment by ptnapoleon** [comment]
This is failing for me locally, but I'm going to try it on cassci

**Comment by ptnapoleon** [comment]
Passes on cassci, so I'm merging.

**Pull request merged**